ui/src/main/webapp/index.js

```diff
@@ -29,13 +31,15 @@ if (process.env.NODE_ENV !== 'production') {
     // use <App /> here rather than require() a <NextApp />.
     try {
       const routes = require('./app').routes
-      render(
+      instance = render(
         <AppContainer errorReporter={({ error }) => { throw error }}>
           <Router history={hashHistory} routes={routes} />
         </AppContainer>,
         document.getElementById('root'))
     } catch (e) {}
   })
+
+  window.forceUpdateThemeing = () => setTimeout(() => deepForceUpdate(instance), 0)
```

Comment:
This is one we ran by @djblue. There was an issue where wrapped components (i.e. the pages that aren't the home page) weren't updating their color as we changed the palettes.
Using deepForceUpdate normally defeats the entire purpose of React, and using it regularly is 👎 , but in our case, it's the only way we were able to live update the color of components that weren't directly receiving the state change information via React's normal channels.
That's my understanding, but we should double-check with @djblue.